DashKit/PodDoseProgressEstimator.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  PodDoseProgressEstimator.swift
+//  DashKit
+//
+//  Created by Pete Schwamb on 5/31/19.
+//  Copyright © 2019 Tidepool. All rights reserved.
+//
+
+import Foundation
+import LoopKit
+
+class PodDoseProgressEstimator: DoseProgressTimerEstimator {
+
+    let dose: DoseEntry
+
+    weak var pumpManager: PumpManager?
+
+    override var progress: DoseProgress {
+        let elapsed = -dose.startDate.timeIntervalSinceNow
+        let duration = dose.endDate.timeIntervalSince(dose.startDate)
+        let percentComplete = min(elapsed / duration, 1)
+        let delivered = pumpManager?.roundToSupportedBolusVolume(units: percentComplete * dose.programmedUnits) ?? dose.programmedUnits
```

Comment:
The part after the ?? should never be executed, but to keep the return value non-optional and still keep our pumpManager reference weak to avoid retain cycles, we need to have something there.